ideascube/serveradmin/wifi.py

```diff
@@ -0,0 +1,183 @@
+from itertools import groupby
+from operator import attrgetter
+from uuid import uuid4
+
+from django.utils.translation import ugettext as _
+
+try:
+    from NetworkManager import (
+        NetworkManager, Settings as NMSettings, NM_DEVICE_TYPE_WIFI)
+
+except Exception as e:
+    import sys
+    sys.stderr.write("Could not import the NetworkManager module: %s\n" % e)
+    sys.stderr.write("Wifi management will not work properly\n")
+    sys.stderr.flush()
+
+    # Set these to None, people will get AttributeError if they try calling
+    # the wifi views, but they have been warned above
+    NetworkManager = None
+    NMSettings = None
+
+    # Hardcode this one, it is a constant anyway in the NetworkManager module,
+    # and hardcoding it allows the tests to run anyway
+    NM_DEVICE_TYPE_WIFI = 2
+
+
+class AvailableWifiNetwork(object):
+    @classmethod
+    def all(cls):
+        result = []
+
+        if not NetworkManager.WirelessEnabled:
+            return result
+
+        for device in NetworkManager.Devices:
+            if device.DeviceType != NM_DEVICE_TYPE_WIFI:
+                continue
+
+            aps = device.SpecificDevice().GetAllAccessPoints()
+            aps = sorted(aps, key=attrgetter('Ssid'))
+
+            for ssid, grouper in groupby(aps, attrgetter('Ssid')):
+                if not ssid:
+                    continue
+
+                # There usually are more than one network for a given SSID
+                # (for example at 2.4 and 5GHz)
+                # Displaying each more than once would confuse users, so let's
+                # only pick the one with the highest frequency.
+                for ap in sorted(grouper, key=attrgetter('Frequency')):
+                    result.append(cls(ap, device))
+                    break
+
+            break
```

Comment:
@fheslouin can we use indistinctly both wifi cards?

> The only issue I can see if is if both device find the same SSIDs, then the list will have things doubled.

Using an OrderedDict may help :)